### ISSUE TITLE
Add EoMT

### ISSUE THREAD

**Issue by NielsRogge** [huggingface/transformers#37171]
### Model description

It'd be great to make the EoMT model available in the Transformers library, because it does image segmentation (instance, semantic and panoptic) in a much simpler way that [Mask2Former](https://huggingface.co/docs/transformers/main/en/model_doc/maskformer) and [OneFormer](https://huggingface.co/docs/transformers/en/model_doc/oneformer). The latter consist of multiple building blocks (vision backbone, pixel decoder, transformer decoder), making them overall pretty complex and hard to implement.

Simplicity allows to inherit the optimizations already done for ViT/BERT to be applied to segmentation as well. 

Initial thread: https://github.com/tue-mps/eomt/issues/1

### Open source status

- [x] The model implementation is available
- [x] The model weights are available

### Provide useful links for the implementation

Paper: https://huggingface.co/papers/2503.19108

Code: https://github.com/tue-mps/eomt

**Comment by NahieliV** [comment]
Hi. I would like to contribute. I am new to open source but I have been using transformers for years. I am a senior data scientist. I would like to implement it :) .

**Comment by NahieliV** [comment]
I just saw that the authors of the model itself will implement the model. I'll look for another issue then.

**Comment by qubvel** [comment]
cc @NiccoloCavagnero @tommiekerssies regarding the integration, please let us know if you have the bandwidth to proceed with it

**Comment by qubvel** [comment]
It would be super nice to have loss function implemented as well, the finetuning routine is usually provided as a separate example script / notebook and can be done on our side or by the community if you have a lack of bandwidth

**Comment by tommiekerssies** [comment]
I'll start with the model for inference, which is the easiest (simply a ViT with some extra query tokens, a class head for the queries, a mask head for the queries, and an upscale head for the patch tokens). Afterwards, we can implement the model for training (which adds masked attention and mask annealing). For the loss, we are actually already using the loss from the Transformers implementation of Mask2Former.

**Comment by qubvel** [comment]
Hey! If anyone from the community is down to continue contributing the model, we would appreciate your help! You can start your work from the linked above PR 🤗

**Comment by yaswanth19** [comment]
HI @qubvel , I can pick this up. Since I already have experience in contributing a model, it would be quicker and easier for me to iterate 🤗 .

**Comment by qubvel** [comment]
@yaswanth19 sounds good 👍

**Comment by tommiekerssies** [comment]
Thanks, @yaswanth19! Let us know if you need anything or have any questions along the way.
